contracts/lpp/src/contract/lender.rs

```diff
@@ -17,18 +17,20 @@ use crate::{
     state::Deposit,
 };
 
-pub(super) fn try_deposit<LPN>(
+pub(super) fn try_deposit<Lpn>(
     deps: DepsMut<'_>,
     env: Env,
     info: MessageInfo,
 ) -> Result<MessageResponse>
 where
-    LPN: 'static + Currency + DeserializeOwned + Serialize,
+    Lpn: 'static + Currency + DeserializeOwned + Serialize,
 {
     let lender_addr = info.sender;
     let amount = bank::received_one(info.funds)?;
 
-    let lpp = LiquidityPool::<LPN>::load(deps.storage)?;
+    let lpp = LiquidityPool::<Lpn>::load(deps.storage)?;
+
+    lpp.check_utilization_rate(&deps.querier, &env)?;
```

Comment:
The decision of whether a deposit would accepted or not should be based on the rule of maintaining the minimum utilization limit. So this check is incorrect. Rather you should compare the amount to the remaining deposit capacity.